Submitted By: Joe Locash <jlocash at gmail dot com>
Date: 2026-06-13
Initial Package Version: 2.3.3
Upstream Status: Not submitted
Origin: Self
Description: Fix building with openssl-4

diff -Nuarp mutt-2.3.3.orig/mutt_ssl.c mutt-2.3.3/mutt_ssl.c
--- mutt-2.3.3.orig/mutt_ssl.c	2026-06-11 22:58:15.000000000 -0400
+++ mutt-2.3.3/mutt_ssl.c	2026-06-13 14:58:24.344683443 -0400
@@ -982,10 +982,10 @@ static int check_host (X509 *x509cert, c
       subj_alt_name = sk_GENERAL_NAME_value(subj_alt_names, i);
       if (subj_alt_name->type == GEN_DNS)
       {
-	if (subj_alt_name->d.ia5->length >= 0 &&
-	    mutt_strlen((char *)subj_alt_name->d.ia5->data) == (size_t)subj_alt_name->d.ia5->length &&
+        if (ASN1_STRING_length(subj_alt_name->d.ia5) >= 0 &&
+            mutt_strlen((char *)ASN1_STRING_get0_data(subj_alt_name->d.ia5)) == (size_t)(ASN1_STRING_length(subj_alt_name->d.ia5)) &&
 	    (match_found = hostname_match(hostname_ascii,
-					  (char *)(subj_alt_name->d.ia5->data))))
+					  (char *)(ASN1_STRING_get0_data(subj_alt_name->d.ia5)))))
 	{
 	  break;
 	}
